src/main/java/uk/gov/hmcts/reform/blobrouter/config/StorageConfiguration.java

```diff
@@ -57,4 +58,14 @@ public static BlobContainerClient getCrimeStorageClient(
             .containerName(containerName)
             .buildClient();
     }
+
+    @Bean("bulk-scan-blob-client-builder")
+    public BlobContainerClientBuilder getBulkScanBlobContainerClientBuilder(
+        HttpClient httpClient,
+        @Value("${storage.bulkscan.url}") String bulkScanStorageUrl
+    ) {
+        return new BlobContainerClientBuilder()
+            .httpClient(httpClient)
+            .endpoint(bulkScanStorageUrl);
+    }
```

Comment:
Is this builder immutable, i.e. will there be no side effects if it's reused?

Reply:
it is thread safe. 
we still create client for each request.

Reply:
We execute a sequence of methods on this builder:
```
                    .sasToken(bulkScanSasTokenCache.getSasToken(containerName))
                    .endpoint(bulkScanStorageUrl)
                    .containerName(containerName)
                    .buildClient();
```

Is it not impossible that these calls will be mixed by two different threads so that when each calls `.buildClient()` they might end up with a setup that they didn't expect?